File: evals/iteration-7-go/eval-4-mutex-misuse/baseline/outputs/counter.go
package counter

import "sync"

// Counter is a thread-safe counter.
type Counter struct {
	sync.Mutex
	count int
}

// Inc increments the counter.
func (c Counter) Inc() {
	c.Lock()
	defer c.Unlock()
	c.count++
}

// Value returns the current counter value.
func (c Counter) Value() int {
	c.Lock()
	defer c.Unlock()
	return c.count
}

File: evals/iteration-5-rust/eval-5-mutex-deadlock/baseline/outputs/counter.rs
use std::sync::Mutex;

pub struct Counter {
    value: Mutex<u64>,
}

impl Counter {
    pub fn new() -> Self {
        Counter {
            value: Mutex::new(0),
        }
    }

    pub fn increment(&self) {
        let mut guard = self.value.lock().unwrap();
        *guard += 1;
    }

    pub fn value(&self) -> u64 {
        *self.value.lock().unwrap()
    }
}

impl Default for Counter {
    fn default() -> Self {
        Self::new()
    }
}
